### PR TITLE
PHOENIX-6022 RVC Offset does not handle trailing nulls properly

### PR DESCRIPTION
**Pull request by dbwong**
https://issues.apache.org/jira/browse/PHOENIX-6022

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/compile/RVCOffsetCompiler.java [hunk 1]
```diff
@@ -240,34 +246,41 @@ public CompiledOffset getRVCOffset(StatementContext context, FilterableStatement
...
 
+        //We do not generate a point lookup today in phoenix if the rowkey has a trailing null, we generate a range scan.
         if (!scanRanges.isPointLookup()) {
-            throw new RowValueConstructorOffsetNotCoercibleException(
-                    "RVC Offset must be a point lookup.");
-        }
-
-
-        RowKeySchema.RowKeySchemaBuilder builder = new RowKeySchema.RowKeySchemaBuilder(columns.size());
+            //Since we use a range scan to garuntee we get only the null value and the upper bound is unset this suffices
```
Comment (yanxinyi) [inline phoenix-core/src/main/java/org/apache/phoenix/compile/RVCOffsetCompiler.java:256]: nit: typo `garuntee`

File: phoenix-core/src/main/java/org/apache/phoenix/compile/RVCOffsetCompiler.java [hunk 2]
```diff
@@ -240,34 +246,41 @@ public CompiledOffset getRVCOffset(StatementContext context, FilterableStatement
             }
         }
 
+        byte[] key;
+
         // check to see if this was a single key expression
         ScanRanges scanRanges = context.getScanRanges();
 
+        //We do not generate a point lookup today in phoenix if the rowkey has a trailing null, we generate a range scan.
         if (!scanRanges.isPointLookup()) {
```
Comment (ChinmaySKulkarni) [inline phoenix-core/src/main/java/org/apache/phoenix/compile/RVCOffsetCompiler.java:255]: I guess pushing the logic for trailing null inside the isPointLookup() API is a bigger change that's too risky?

File: phoenix-core/src/main/java/org/apache/phoenix/compile/RVCOffsetCompiler.java (hunk 1 above)
Comment (dbwong) [reply phoenix-core/src/main/java/org/apache/phoenix/compile/RVCOffsetCompiler.java]: done

**Comment by gjacoby126** [comment]
JIRA already committed, closing

**Pull request closed without merging**